tests/replace.rs

```diff
@@ -0,0 +1,10 @@
+mod common;
+
+#[test]
+fn replace() {
+    let regex = common::regex(
+        r"\b([sS])uc(?:cs|s?)e(ed(?:ed|ing|s?)|ss(?:es|ful(?:ly)?|i(?:ons?|ve(?:ly)?)|ors?)?)\b",
+    );
+    let result = regex.replace("a sucessful b", "${1}ucce$2");
+    assert_eq!(result, "a successful b");
```

Comment:
Can you add some more tests? It looks like there's none for `replace_all` (not even doctests). Also not sure if all the replacers are covered, it would be good to have coverage of all of them.